### DISCUSSION TITLE
Can't join forwarder to grid from different subnet

### DISCUSSION THREAD

**Discussion by Todd998** [Security-Onion-Solutions/securityonion#8836] [General] [answered]
Hi all, I've seen several threads with the same problem I'm having but I haven't seen a solution. I'm running 2.3.160 for my manger searcher and on two forward sensors. Both sensors connected to the manager really easy but then I need to move one to another subnet. All of our setup is on-prem with the manger and one node in the same subnet and a node in another subnet. All were installed by installing Centos 7 first, then pulling security onion from git as described in the docs.

After moving the one to a new subnet, I tried giving it a new IP and rerunning setup but it didn't work so I rebuilt it including reinstalling Centos. I tried joining it to the grid from the other subnet but it fails with the error "Attempt to authenticate with the salt master failed with timeout error." 

Searching /opt/so/log/salt/master showed no reference to the new node.

I've verified there are no firewall policies between the manger and forwarder and went as far as turning off the Centos firewall on the forwarder, temporarily. 

I'm not sure what to check next. On thread said to manually add the ip of the forwarder to the manger. When I ran that command it said the entry already existed (this was after a couple of attempts at joining the grid.

Thank you for any help you have.

**Comment by ItzDeadShot** [comment]
I had a similar problem a few weeks ago but I was connecting the manager and forwarder in two different private networks. So in my situation, I had my manager node in 10.207.160.0/23 network and forwarder node in 192.168.2.0/24 network. I will list down the steps I took to connect them together.

The very first thing I had to do was create a NAT on the router near my forward node to translate the internal network (192 net) to an external IP (10 net). Correct me if I am wrong but it might not be an issue for you as you are trying to connect nodes between subnet.

Then I check the connectivity between my sensor node and manager node by simply pinging manager from my sensor node, if the subnets are configured properly then it should not be a problem but just in case.

Now you need to make sure that the ports for salt (4505 and 4506) are enabled on the firewall that is between your forward node and manager node. And also add the forward host IP to the firewall in the firewall configuration of your manager node.  You can check the connectivity to these ports using the command "nc -z -v <manager_ip> 4505" and "nc -z -v <manager_ip> 4506".

After all that was done, I was able to connect the manager to the forward node in two different networks.

If you have specific errors, from the setup logs , you can share them so I can see if I got those as well and share how I solved them.

P.S.: I am new to SOS so my solution/answer might not be definitive.

**Reply by Todd998** [reply]
@ItzDeadShot, the nc command didn't work to ports 4505 and 4506 from the sensor to the manager. They returned the error, no route to host. According to this article, https://unix.stackexchange.com/questions/353452/no-route-to-host-with-nc-but-can-ping that is a firewall response so I disabled the rule on the manager,

`INPUT rule, REJECT     all  --  *      *       0.0.0.0/0            0.0.0.0/0            reject-with icmp-host-prohibited`

 I also verified that there were INPUT rules on the manager allowing my sensor but for whatever reason the traffic was triggering the reject rule instead. 

`num   pkts bytes target     prot opt in     out     source               destination
1        0     0 ACCEPT     tcp  --  *      *       xxx.xxx.xx.9         0.0.0.0/0            tcp dpt:4506
2        0     0 ACCEPT     tcp  --  *      *       xxx.xxx.xx.9         0.0.0.0/0            tcp dpt:4505
`

After that nc timed out. Progress. For testing purposes, I turned off the firewall on the manager. I ran so-setup-network from the sensor again and this time it finished. It had errors and took an extraordinary amount of time but it finally showed up in the Grid however, the process status is unknown and I doubt it is working properly.

After that bit of testing it seems the problem is with the firewall. It probably has something to do with not using the ISO and installing CentOS 7 manually.

Progress but still not resolved.

**Reply by ItzDeadShot** [reply]
Is there a reason for you to manually install SO on CentOS 7 rather than using the ISO? You can try to use the ISO and see if the issue still exists. 
As for the process status, did you see all the services running in your forward node showing [OK] when you run `sudo so-status` in your forward node? If there are some errors, then what is the output of `sudo salt-call state.highstate`?

**Comment by innovate-support** [comment]
Long thread, but worth the read.  I think it may be the issue you're having.
https://github.com/Security-Onion-Solutions/securityonion/discussions/2025

**Reply by Todd998** [reply]
Thank you @innovate-support, I spent all day yesterday working through that. It was good learning time but it didn't resolve the problem.

**Comment by Todd998** [comment]
We tried the ISO but it wouldn't work on any of the 3 boxes. It would boot but shortly after we'd get a blinking cursor. We could Ctrl+Alt+Del and reboot but never got an error. The documentation suggested this route so we did it but maybe we should back up and try the ISO again. I feel like I wouldn't be having these problems if that had worked.

It is odd that the first sensor we built works fine and the second does not. The second works fine if we install it on the same subnet as the manager but not where it is.

Here are the results of those commands:
so-nginix --- waiting to start
so-sensoroni ---- waiting to start
so-steno ----- waiting to start
so-suricata --- waiting to start
so-zeek ------- waiting to start

Docker---- OK

salt-call state.highstate:

`local:
    Data failed to compile:
    The function "state.highstate" is running as PID 2685 and was started at  with jid req
`

**Comment by dougburks** [comment]
> After moving the one to a new subnet, I tried giving it a new IP and rerunning setup but it didn't work so I rebuilt it including reinstalling Centos. I tried joining it to the grid from the other subnet but it fails with the error "Attempt to authenticate with the salt master failed with timeout error."

Please double-check that there are no firewalls or other devices that may blocking communication on the ports listed at https://docs.securityonion.net/en/2.3/firewall.html#node-communication.

**Reply by Todd998** [reply]
There are no firewalls between them except for the ones on the systems themselves. Disabling the firewall on the manager did allow the sensor to complete the setup however it still doesn't operate properly. There errors are posted above.

**Comment by Todd998** [comment, marked as answer]
I have verified that I can connect from the node to all the documented ports on the manager as long as the manager firewall is down. I also verified that the sensor is in the sensor and minions section of /opt/so/saltstack/local/salt/firewall/hostgroups.local.yaml

**Reply by ItzDeadShot** [reply]
Can you confirm that the packets your manager is receiving have the IP of the sensor node, or is it the IP of the intermediate that is forwarding the packets from your sensor node to the manager node?

You can confirm this by running `tcpdump -i <manager_interface> icmp` on your manager node and then ping the manager node from your sensor node. If there are another IP making ICMP requests for your sensor node, then add that IP to your manager node firewall.

As for the issue with so-status, you may check salt logs in `/opt/so/log/salt/` for any errors and post them here. The error that I got when the status was stuck at WAITING_TO_START was something to do with the manager hostname, which I solved by adding a new line `<manager_hostname> <manager_IP>` manually in /etc/hosts in the sensor node.

Hope this helps.

**Reply by Todd998** [reply]
Thanks @ItzDeadShot We got it. It was a routing issue. We identified it by using tcpdump. It showed packets coming from the gateway and not the host. NAT was enabled in the policy. Turning it off fixed it. 

Thanks for all the help everybody!
